fred/cogs/levelling.py

```diff
@@ -4,36 +4,43 @@
 import math
 from datetime import *
 
-import nextcord.ext.commands as commands
-from nextcord import DMChannel
+from nextcord import DMChannel, Message, Guild
+from nextcord.ext.commands import MemberNotFound
 
-import config
-from libraries import common
+from .. import config
+from ..libraries import common
+
+logger = common.new_logger(__name__)
 
 
 class UserProfile:
-    def __init__(self, user_id, guild, bot):
+    def __init__(self, user_id: int, guild: Guild):
         self.guild = guild
-        self.bot = bot
         self.user_id = user_id
-        self.logger = logging.Logger("USERPROFILE")
 
-        assert bot.intents.members, "The bot has no member intents, so it cannot do levelling!!"
-        self.member = bot.get_user(user_id)
-        assert self.member is not None, "This member does not exist o.0"
-        self.logger.info(f"Found member id {self.member}")
+        self.member = guild.get_member(user_id)
+        if self.member is None:
+            logger.warning(f"Unable to retrieve information about user {user_id}")
+            raise MemberNotFound(f"Unable to retrieve information about user {user_id}")
+
+        logger.info(f"Found member id {self.member}")
 
         if DB_user := config.Users.fetch(user_id):
             self.DB_user = DB_user
         else:
-            self.DB_user = config.Users(user_id=user_id, full_name=f"{self.member.name}#{self.member.discriminator}")
+            self.DB_user = config.Users(user_id=user_id)
+
+    @property
+    def rank(self):
+        return self.DB_user.rank
```

Comment:
This does a DB call instead of the previous way which didn't. Not really an issue here tbh but just a reminder, probably not worth it to add more calls for more pythonic code imo

Reply:
We could use functools.cachedproperty

Reply:
Yeah that would be nicer. Still not sold on over-complicating this when it had no issue but it won't really hurt

Reply:
Seems a bit weird to use so I'll do something better